lib/CL/devices/pthread/pthread_scheduler.c

```diff
@@ -336,12 +337,12 @@ work_group_scheduler (kernel_run_command *k,
   pocl_free_kernel_arg_array_with_locals ((void **)arguments,
                                           (void **)arguments2, k);
 
-  return 1;
+  POCL_ATOMIC_ADD (k->execution_failed, execution_failed);
```

Comment:
Could this addition overflow and, by a very unlikely chance, turn an execution failure to be recorded as successful one?

Reply:
Yes in theory it could overflow. I'll change it to atomic_or